[CIS-1811] Add Extra Data Usage Improvements (#2174)
GetStream/stream-chat-swift: docusaurus/docs/iOS/client/extra-data.md

```swift
let currentUserController = client.currentUserController()
currentUserController.updateUserData(
    name: "John Doe",
    imageURL: nil,
    userExtraData: ["email": .string("john.doe@example.com")],
    completion: nil
)
```

For a more complete example now, let's imagine you want to add a ticket information to a message.

```swift
let extraData: [String: RawJSON] = [
    "ticket": .dictionary([
        "name": .string("Rock Concert"),
        "price": .double(20)
    ])
]
let channelController = client.channelController(for: yourChannelId)
channelController.createNewMessage(text: "A new message", extraData: extraData)
```

:::tip
Since `RawJSON` implements `ExpressibleByDictionaryLiteral`, you can simplify the example above like this:
```swift
let extraData: [String: RawJSON] = [
    "ticket": [
        "name": .string("Rock Concert"),
        "price": .double(20)
    ])
]
```
***Note:*** This is only available on 4.19.0+ versions of the SDK.
:::

## Reading Extra Data

All of the most important domain models in the SDK have a `extraData` property that you can read the additional information that is added.

> Since the 4.19.0 version of the SDK you can read extra data properties very easily. The following code snippet shows how to get an email from a user's extra data.

Author: ```suggestion
Since the 4.19.0 version of the SDK, you can read extra data properties very easily. The following code snippet shows how to get an email from a user's extra data.
```

let name = message.extraData["ticket"]?["name"]?.stringValue ?? ""
let price = message.extraData["ticket"]?["price"]?.doubleValue ?? 0.0
```

As you can see above, each type of value can be easily accessible from an extra data property. The SDK will try to convert the raw type to a strongly typed value and return it if the property exists, and if the type is correct. Below is the list of all values supported:

- `stringValue: String?`
- `numberValue: Double?`
- `boolValue: Bool?`
- `dictionaryValue: [String: RawJSON]?`
- `arrayValue: [RawJSON]?`
- `stringArrayValue: [String]?`
- `numberArrayValue: [Double]?`
- `boolArrayValue: [Bool]?`

## Advanced Example

Most likely your app has more complex data structures compared to the ones described above. So, let's see an example of how you could map your domain models to extra data and vice-versa by imagining that a message can have details of a booking flight.

```swift
struct BookingFlight {
    let flightNumber: Double
    let departureDate: Date
    let arrivalDate: Date
    let price: Double
    let passengers: [Passenger]
    let destinations: [String]
}

struct Passenger {
    let name: String
    let age: Int
}
```

> From the example above, now let's see how we can provide Extra Data mappings for these models:

Author: ```suggestion
Next, let's see how we can provide Extra Data mappings for these models:
```
